### PR TITLE
Banning contests 1308, 1309, 1431 and 1432

### PR DESCRIPTION
**Pull request by dgcnz**
I'm filtering the contests described in issue #324 (and 1431, 1432) only at `update_task` and `monitor_task` so that one can still query for them. 

For now it's basically a global list `CONTEST_BLACKLIST` with the banned contest_ids, but I think we could also automate it.

As far as I could tell there are 3 kinds of "unrated" contests based on their result on [contest.ratingChanges](https://codeforces.com/api/contest.ratingChanges?contestId=contestid): 

1. Normal CF contests that were unrated: `{"status":"OK","result":[]}`
2. Non-CF contests that should not be monitored (this was the case in the aforementioned banned contests but maybe there are false-positives): `{"status":"FAILED","comment":"contestId: Rating changes are unavailable for this contest"}` 
3. Normal CF contests that haven't finished yet (I tested this with contest 1438, so this might also not be the case in actual running contests): `{"status":"FAILED","comment":"contestId: Rating changes are unavailable, because the contest isn\u0027t finished yet"}`

The code in `ratingChanges` at `codeforces_api.py:282` deals with errors in this way:
```python
        except TrueApiError as e:
            if 'not found' in e.comment:
                raise ContestNotFoundError(e.comment, contest_id)
            if 'Rating changes are unavailable' in e.comment:
                raise RatingChangesUnavailableError(e.comment, contest_id)
            raise
```

We could maybe add another if-clause there to deal specifically with the banned contests instead of manually adding them on the code (I'm fairly new to TLE's code so I didn't dare to touch more code haha). 

Nonetheless, the current change should work in the meantime.

### CODE REVIEW

**Comment by aryanc403** [comment]
https://discordapp.com/invite/2CJ6qvY

**Comment by dgcnz** [comment]
Thanks, I just joined :)

File: tle/util/cache_system2.py [hunk 1]
```diff
@@ -431,8 +434,10 @@ async def _update_task(self, _):
     @tasks.task_spec(name='RatingChangesCacheUpdate.MonitorNewlyFinishedContests',
                      waiter=tasks.Waiter.fixed_delay(_RELOAD_DELAY))
     async def _monitor_task(self, _):
+
+        check = lambda contest : self.is_newly_finished_without_rating_changes(contest) and contest.id not in CONTEST_BLACKLIST
```
Comment (algmyr) [inline tle/util/cache_system2.py:438]: Again, inline check.

**Review by algmyr — CHANGES_REQUESTED** [review]

File: tle/util/cache_system2.py [hunk 2]
```diff
@@ -431,8 +437,10 @@ async def _update_task(self, _):
     @tasks.task_spec(name='RatingChangesCacheUpdate.MonitorNewlyFinishedContests',
                      waiter=tasks.Waiter.fixed_delay(_RELOAD_DELAY))
     async def _monitor_task(self, _):
-        self.monitored_contests = [contest for contest in self.monitored_contests
-                                   if self.is_newly_finished_without_rating_changes(contest)]
+        self.monitored_contests = [contest for contest in self.monitored_contests if
+                                   self.is_newly_finished_without_rating_changes(contest) and 
+                                   not _is_blacklisted(contest.id)]
```
Comment (algmyr) [inline tle/util/cache_system2.py:442]:
```suggestion
        self.monitored_contests = [
            contest for contest in self.monitored_contests
            if self.is_newly_finished_without_rating_changes(contest)
            and not _is_blacklisted(contest.id)
        ]
```

File: tle/util/cache_system2.py [hunk 3]
```diff
@@ -416,8 +420,10 @@ async def _update_task(self, _):
         # A contest also has empty list if it is unrated. We assume that is the case if
         # _RATED_DELAY time has passed since the contest end.
 
-        to_monitor = [contest for contest in self.cache_master.contest_cache.contests_by_phase['FINISHED']
-                      if self.is_newly_finished_without_rating_changes(contest)]
+        to_monitor = [contest for contest in self.cache_master.contest_cache.contests_by_phase['FINISHED'] 
+                      if self.is_newly_finished_without_rating_changes(contest) and
+                      not _is_blacklisted(contest.id)]
```
Comment (algmyr) [inline tle/util/cache_system2.py:425]:
```suggestion
        to_monitor = [
            contest for contest in
            self.cache_master.contest_cache.contests_by_phase['FINISHED']
            if self.is_newly_finished_without_rating_changes(contest)
            and not _is_blacklisted(contest.id)
        ]
```

**Review by algmyr — CHANGES_REQUESTED** [review]

**Review by algmyr — APPROVED** [review]
LGTM now :)

**Pull request merged**